### PR TITLE
feat(core): allow user to provide docker --security-opt when bundling

### LINKED ISSUE #14681: (core): Allow user to provide --security-opt to DockerRunOptions
Allow user to provide `--security-opt` within [DockerRunOptions](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.DockerRunOptions.html)

### Use Case

In restrictive development environments, users may need to include [Security Configurations](https://docs.docker.com/engine/reference/run/#security-configuration) for all `docker run` commands.

If a developers org provides development desktops that allow `docker` commands, but requires specific `--security-opt` flags when running, they're forced to build their docker assets manually, and point to the built zip file

### Proposed Solution

Allow an optional `securityOpt` flag for [DockerRunOptions](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.DockerRunOptions.html)

### Other

PR #14681 

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by maafk**
Allows user to pass `securityOpts` to [DockerRunOptions](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.DockerRunOptions.html) so that docker [security configurations](https://docs.docker.com/engine/reference/run/#security-configuration) can be set for developers working in restrictive dev environments.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by maafk** [comment]
@rix0rrr First time contributor here. 

Can I please get an approval to run a workflow?

Thanks!

**Review by rix0rrr — DISMISSED** [review]

**Review by rix0rrr — DISMISSED** [review]

**Review by rix0rrr — DISMISSED** [review]

**Review by rix0rrr — APPROVED** [review]

**Comment by rix0rrr** [comment]
Please don't merge from master. It will just dismiss any approval.

**Review by rix0rrr — APPROVED** [review]

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**
